app/modules/tmr.c

```diff
@@ -60,6 +60,10 @@ tmr.softwd(int)
 #define TIMER_MODE_AUTO 1
 #define TIMER_IDLE_FLAG (1<<7) 
 
+// assuming system_timer_reinit() has *not* been called
+static const uint32_t MAX_TIMEOUT = 0x68D7A3; //SDK 1.5.3 limit
+static const char* MAX_TIMEOUT_STR = "Range: 1-6870947";
```

Comment:
How about deducing 6870947 in `MAX_TIMEOUT_STR` from `MAX_TIMEOUT` just above? You'd eliminate a potential pitfall down the road when the max timeout value changes (should you forget to update the error message at the same time).